### PR TITLE
Define JSON schemas for weapons

### PR DESCRIPTION
**Pull request by jcuratolo**
Defined schemas for existing weapon types.

### CODE REVIEW

File: assets/weapons/tags/schemas/base_weapon.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            "type": "number",
+            "description": "The rate of fire in units of rounds per second"
+        },
+        "bullets": {
+            "type": "integer",
+            "description": "The number of rounds per magazine"
+        },
+        "ammo": {
+            "type": "integer",
+            "description": "TODO"
```
Comment (jcuratolo) [inline assets/weapons/tags/schemas/base_weapon.json:33]: Anything with a TODO description is something I wasn't sure of. I didn't dig around to infer usage so we could discuss first.

File: assets/weapons/tags/schemas/base_weapon.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        },
+        "damage": {
+            "type": "integer",
+            "description": "The amount of damage applied to the target"
+        },
+        "reloadSpeed": {
+            "type": "number",
+            "description": "The time required to reload the weapon in seconds"
+        },
+        "randomness": {
```
Comment (jcuratolo) [inline assets/weapons/tags/schemas/base_weapon.json:53]: After finalizing this we can probably convert this to just a boolean?

File: assets/weapons/tags/schemas/base_weapon.json [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            ],
+            "description": "TODO"
+        },
+        "spreadPattern": {
+            "type": "array",
+            "items": {
+                "type": "array",
+                "items": {
+                    "type": "number",
+                    "minimum": -1,
```
Comment (jcuratolo) [inline assets/weapons/tags/schemas/base_weapon.json:72]: Assumed that these are just supposed to be unit vectors, let me know if otherwise

**Comment by jcuratolo** [comment]
Does Godot have utilities to consume and apply JSON schemas? Might be nice to verify on startup or commit etc

File: assets/weapons/tags/schemas/base_weapon.json (hunk 3 above)
Comment (yaelatletl) [reply assets/weapons/tags/schemas/base_weapon.json:72]: It whould be an array with pairs of spread coords, I think going further than 1/-1 is possible, but keeping it clamped might be a better practice

**Review by yaelatletl — APPROVED** [review]
Alright, all good, we can handle all those TODO later on

**Pull request merged**